src/styles/styleRules.ts

```diff
@@ -173,8 +173,19 @@ export class StyleRules {
                 styleRule.input[1]
               } ${StyleRules.sanitizeValue(styleRule.type, styleRule.value)}`
             : `${StyleRules.getTypeLabel(styleRule.itemType)}`;
-          const value = styleRule.style.image;
-          StyleRules.updateLegend(result, {label: label, value: value});
+          const imageStyle = styleRule.style.image as IStyleImage;
+          if (
+            imageStyle.url &&
+            typeof imageStyle.url === 'object' &&
+            imageStyle.url.type === 'data'
+          ) {
+            // If the custom icon is read from a node property-value dynamically, then we
```

Comment:
missing a continue here.
Or we check the opposite of the condition to avoid an if else

Reply:
there is not other statement after in the loop so the "continue" would be redundant, do you prefer to add it anyway?

Reply:
ping @aminerhanemi

Reply:
oky good for me